src/main/java/org/sysc4806/sysc4806_group20/Controller/TopicRestController.java

```diff
@@ -4,17 +4,25 @@
 import org.springframework.data.rest.webmvc.ResourceNotFoundException;
 import org.springframework.web.bind.annotation.*;
 import org.sysc4806.sysc4806_group20.Model.Professor;
+import org.sysc4806.sysc4806_group20.Model.Student;
 import org.sysc4806.sysc4806_group20.Model.Topic;
 import org.sysc4806.sysc4806_group20.Service.ProfessorService;
+import org.sysc4806.sysc4806_group20.Service.StudentService;
 import org.sysc4806.sysc4806_group20.Service.TopicService;
 
 @RestController
 @RequestMapping("/api/topics")
 public class TopicRestController {
-    @Autowired
+
     private TopicService topicService;
-    @Autowired
     private ProfessorService professorService;
+    private StudentService studentService;
+
+    TopicRestController(TopicService topicService, ProfessorService professorService, StudentService studentService){
```

Comment:
You don't need a constructor here

Reply:
Ya but, instead of auto wiring each field with the annotation its cleaner to do constructor injection.

Reply:
I don't see the point, Auto wiring will take care of the stuff, you won't be calling the constructor anywhere